frontend/src/pages/signin-page.tsx

```diff
@@ -0,0 +1,106 @@
+import { Link, useNavigate } from 'react-router-dom';
+import AddGoogleIcon from '@/assets/svg/google-color-icon.svg';
+import AddGithubIcon from '@/assets/svg/github-icon.svg';
+import { useForm } from 'react-hook-form';
+import type { FieldValues } from 'react-hook-form';
+import { TSignInSchema, signInSchema } from '@/lib/types';
+import 'react-toastify/dist/ReactToastify.css';
+import { zodResolver } from '@hookform/resolvers/zod';
+import { toast } from 'react-toastify';
+
+function signin() {
+  const navigate = useNavigate();
+
+  const { register, 
+        handleSubmit, 
+        formState: { errors , isSubmitting},
+        reset
+      } = useForm<TSignInSchema>({resolver: zodResolver(signInSchema)});
+
+  const onSubmit = async (data: FieldValues) => {
+
+    if(data.email === "abc@gamil.com"){
+      toast.error('Submitting form is failed');
+      return;
+    }
+    
+    // TODO: Server-side validation
+    await new Promise((resolve) => setTimeout(resolve, 1000));
+    
+    reset();
+    navigate("/");
+  }
+
+  return (
+    <div className="flex-grow cursor-default bg-white m-4 py-4">
+      <div className="mb-4 flex justify-center">
+        <div className="flex items-center justify-center w-full">
+          <h2 className="text-center text-22 sm:text-2xl font-bold text-black w-3/4">
+            Sign in to WanderLust
```

Comment:
Lets use text-xl here for mobile devices

Reply:
Okay.

Reply:
Done.

Reply:
Done.